### PR TITLE
[DRWTSN32] Improve stack dumping

### PR DESCRIPTION
**Pull request by learn-more**
Especially when dealing with nullpointers.

### CODE REVIEW

File: base/applications/drwtsn32/stacktrace.cpp [hunk 1]
```diff
@@ -59,23 +59,28 @@ void PrintStackBacktrace(FILE* output, DumpData& data, ThreadData& thread)
...
-        if (!StackFrame.AddrPC.Offset)
-            break;
-
-        if (LastFrame >= StackFrame.AddrFrame.Offset)
-            break;
-
-        LastFrame = StackFrame.AddrFrame.Offset;
+        if (StackFrame.AddrPC.Offset == StackFrame.AddrReturn.Offset)
+        {
+            if (RecursionDepth++ > 50)
```
Comment (HBelusca) [inline base/applications/drwtsn32/stacktrace.cpp]: Maybe define this magic value of "50" somewhere at the beginning of the code?

File: base/applications/drwtsn32/stacktrace.cpp [hunk 2]
```diff
@@ -59,23 +61,28 @@ void PrintStackBacktrace(FILE* output, DumpData& data, ThreadData& thread)
...
-    ULONG_PTR LastFrame = StackFrame.AddrFrame.Offset - 8;
-    while(StackWalk64(MachineType, data.ProcessHandle, thread.Handle, &StackFrame, &thread.Context,
-                         NULL, SymFunctionTableAccess64, SymGetModuleBase64, NULL))
+    while (StackWalk64(MachineType, data.ProcessHandle, thread.Handle, &StackFrame, &thread.Context,
+                       NULL, SymFunctionTableAccess64, SymGetModuleBase64, NULL))
     {
-        if (!StackFrame.AddrPC.Offset)
-            break;
-
-        if (LastFrame >= StackFrame.AddrFrame.Offset)
```
Comment (ThFabba) [inline base/applications/drwtsn32/stacktrace.cpp]: Dropping this check intentionally?

Comment (learn-more) [reply base/applications/drwtsn32/stacktrace.cpp]: Yes.

Comment (learn-more) [reply base/applications/drwtsn32/stacktrace.cpp]: The previous checks were too strict, and thus we were not showing a callstack when it was still possible to show something.

**Pull request merged**